### ISSUE TITLE
Fatal compiler error for undeclared identifier

### ISSUE THREAD

**Issue by benharsh** [chapel-lang/chapel#5889]
### Summary of Problem
The following program fails with an internal compiler error, instead of a nice explanatory message for the user:
```chpl
var x = asdf; // asdf is undeclared!
writeln(x);
```
We had the nice "undeclared (first use this function)" message in 1.14, but this internal error is new in 1.15: 
```
undeclared.chpl:1: internal error: NOR1893 chpl Version 1.15.0.54108ba
Note: This source location is a guess.

blah blah blah
```

### Steps to Reproduce

**Associated Future Test(s):**
See test added in #5888

**Comment by benharsh** [comment]
@noakesmichael : this may be a result of recent normalize refactoring. I haven't bisected to figure out which commit changed the behavior.

We really need to ensure all our compiler error messages are tested.

**Comment by benharsh** [comment]
Mike fixed this in #5891, thanks!
